### DISCUSSION TITLE
如何完全丢弃AAAA解析？

### DISCUSSION THREAD

**Discussion by yunyuyuan** [IrineSistiana/mosdns#686] [General]
需要达到的效果就像dnsmasq的`address=/openai.com/::`，我想在mosdns配置所有和dns相关的，就懒得再在dnsmasq里配置一次了。
在mosdns里`- exec: prefer_ipv4`，貌似对客户端是没有效果的，客户端依然会ipv6优先，所以需要类似这样的逻辑：
```
- matches:
    - qname openai.com
  exec:
    #- prefer_ipv4
    - only_ipv4
- ...
- exec: forward 8.8.8.8
```
在查询openai.com时，直接把AAAA的解析丢掉。

**Comment by BozhanL** [comment]
试试用 [qtype](https://irine-sistiana.gitbook.io/mosdns-wiki/mosdns-v5/ru-he-pei-zhi-mosdns/sequence-cha-jian#qtype) 匹配 ```AAAA``` 请求， 然后丢弃或者返回空地址

**Reply by BozhanL** [reply]
似乎 [black_hole](https://irine-sistiana.gitbook.io/mosdns-wiki/mosdns-v5/ru-he-pei-zhi-mosdns/sequence-cha-jian#black_hole) 只配置 ```ipv6``` 地址也能起到相同的效果
